src/Calypso-SystemTools-Core/ClyBrowserMorph.extension.st

```diff
@@ -105,12 +87,61 @@ ClyBrowserMorph >> confirmUnusedVariablesInDefiningClass: variables [
 ClyBrowserMorph >> decorateMethodEditor: aMethodEditor [
 ]
 
+{ #category : #'*Calypso-SystemTools-Core' }
+ClyBrowserMorph >> hasReferencesToClasses: classes [
+	
+	| query references |
+	
+	query := (ClyClassReferencesQuery toAny: classes from: self systemScope).
+	references := query execute.	
+	^(references notEmpty and: [
+		((references items collect: [:each | each origin])
+			difference: (classes flatCollect: [ :each | { each . each classSide } ])) notEmpty])
```

Comment:
I also thought to extract this logic into ClyClassReferencesQuery itself (or special subclass). 
- query ignoreSelfReferences
- ClyExternalClassReferencesQuery